### ISSUE TITLE
Tests should not rely on execution order

### ISSUE THREAD

**Issue by m-vo** [contao/contao#2316]
I sometimes get errors executing our tests - not as a suite but when running single classes/cases. There are some hidden dependencies between test cases or in other words: the execution order matters.

I think we should try to remove those dependencies: It can make writing new tests frustrating, because you now have to execute more then necessary to see if everything's green or not and you're possibly interfering with the original order creating new issues in seemingly unrelated tests. 

**How to reproduce:**

This yields no errors:
```bash
vendor/bin/phpunit --colors=always
```

But this will report over 200 errors. Most of them due to a missing `request_stack` service in the container. 
```bash
vendor/bin/phpunit --colors=always --order-by=random --random-order-seed=1 
```

Example of a single file causing errors when not tested in context:
```bash
vendor/bin/phpunit --colors=always core-bundle/tests/Controller/ContentElement/ContentElementControllerTest.php 
```

**Comment by leofeyer** [comment]
Which Contao versions are affected?

**Comment by m-vo** [comment]
I only tested it with the current master so far but I assume 4.4+ is affected.

**Comment by ausi** [comment]
If it requires some amount of work, I don’t think we need to fix this in the 4.4 test suite.

**Comment by leofeyer** [comment]
Contao 4.4 uses PHPUnit 5.7.27, which does not even support the `--order-by` flag, therefore I guess I will assign this issue to the `4.9` milestone.

**Comment by m-vo** [comment]
I'm working on this. 

I will probably provide multiple PRs because of the various (rather independent) issues.

**Comment by leofeyer** [comment]
Fixed in #4120.
